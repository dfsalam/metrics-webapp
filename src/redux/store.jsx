import { configureStore } from '@reduxjs/toolkit';
import townsReducer from './towns/townsSlice';
import polutionReducer from './polution/polutionSlice';

const store = configureStore({
  reducer: {
    towns: townsReducer,
    polution: polutionReducer,
  },
});

export default store;
