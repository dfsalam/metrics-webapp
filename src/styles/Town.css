.town-card {
  display: flex;
  flex-direction: column;
  width: 50%;
  background-color: rgb(193, 60, 111);
}

.town-card-darker {
  display: flex;
  flex-direction: column;
  width: 50%;
  background-color: rgb(233, 73, 135);
}

.town-button {
  margin-top: 5px;
  background-color: inherit;
  border: 0;
  padding: 0;
  align-self: flex-end;
}

.town-text {
  font-family: 'Gill Sans', serif;
  color: white;
  margin: 2px 5px 7px 0;
  padding-bottom: 0;  
  text-align: right;
}

.arrow {
  font-variation-settings: 'FILL' 0;
}
