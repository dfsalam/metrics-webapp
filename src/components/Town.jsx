import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { useAppDispatch } from '../redux/redux-hooks';
import { getPolution } from '../redux/polution/polutionSlice';
import '../styles/Town.css';

const Town = ({ townId, townName, darker }) => {
  const dispatch = useAppDispatch();
  const classTag = darker ? 'town-card-darker' : 'town-card';
  const path = `details/${townName}`;
  const handler = (name) => {
    dispatch(getPolution(name));
  };
  return (
    <div className={classTag}>
      <button
        className="town-button"
        type="button"
        id={townId}
        onClick={() => handler(townName)}
      >
        <NavLink to={path}>
          <span className="material-symbols-outlined arrow">
            arrow_circle_right
          </span>
        </NavLink>
      </button>
      <h2 className="town-text">{townName.toUpperCase()}</h2>
      <h3 className="town-text">
        ID:
        {townId}
      </h3>
    </div>

  );
};

Town.propTypes = {
  townId: PropTypes.string.isRequired,
  townName: PropTypes.string.isRequired,
  darker: PropTypes.bool.isRequired,
};

export default Town;
