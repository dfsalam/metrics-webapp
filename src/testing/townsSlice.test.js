import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Town from '../components/Town';
import store from '../redux/store';

const mockStore = configureStore([]);

describe('Towns redux state tests', () => {
  it('Expect to initially set towns to an empty object', () => {
    const state = store.getState().towns;
    const initialState = {
      towns: [], isLoading: true, townsSearch: [], searching: false,
    };
    expect(state).toEqual(initialState);
  });
  it('Expect to display towns data from mockStore', () => {
    const towns = [
      {
        region: 'Región Centro Oriente',
        c_digo_dane_del_departamento: '15',
        departamento: 'Boyacá',
        c_digo_dane_del_municipio: '15.476',
        municipio: 'Motavita',
      },
    ];

    const state = {
      towns: {
        towns,
        townsSearch: [],
        isLoading: true,
        searching: false,
      },
    };
    const store = mockStore(state);
    render(
      <Provider store={store}>
        <BrowserRouter><Town townId="15.476" townName="Motavita" /></BrowserRouter>
      </Provider>,
    );
    expect(screen.getByText('Motavita')).toBeDefined();
    expect(store.getState).toMatchSnapshot();
  });
});
