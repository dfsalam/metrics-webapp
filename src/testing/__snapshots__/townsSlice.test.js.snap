// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Towns redux state tests Expect to display towns data from mockStore 1`] = `[Function]`;
